File: benches/zig/fib.zig
// Zig twin of benches/c/fib.c — recursive fib(30)
// Build: zig build-exe -OReleaseFast -femit-bin=build/fib fib.zig
const std = @import("std");

fn fib(n: i32) i32 {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

fn nowUs() i64 {
    return @divTrunc(@as(i64, @intCast(std.time.nanoTimestamp())), 1000);
}

pub fn main() !void {
    _ = fib(20);
    const t0 = nowUs();
    const r = fib(30);
    const t1 = nowUs();
    if (r < 0) return error.Unreachable;
    const out = std.io.getStdOut().writer();
    try out.print("BENCH fib30 iters=1 total_us={d} us_per_op={d}\n", .{ t1 - t0, t1 - t0 });
}

File: benches/nim/fib.nim
## Nim twin of benches/c/fib.c — recursive fib(30)
import std/[times, strformat]

proc fib(n: int): int =
  if n <= 1: return n
  fib(n - 1) + fib(n - 2)

proc nowUs(): int64 =
  int64(epochTime() * 1_000_000.0)

discard fib(20)
let t0 = nowUs()
let r = fib(30)
let t1 = nowUs()
if r < 0: quit(1)
echo &"BENCH fib30 iters=1 total_us={t1 - t0} us_per_op={t1 - t0}"
